src/linesensor.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include <vector>
+#include <atomic>
+
+#include <driver/spi_master.h>
+
+namespace mcp3008 {
+
+/**
+ * \brief The LineSensor device - one MCP3008 chip.
+ *        It is not thread-safe, you have to make sure the methods are called
+ *        from one thread at a time only.
+ *        The @{install} method has to be called before you can use any other methods.
+ */
+class LineSensor {
+public:
+    static constexpr int CHANNELS = 8; //!< Amount of channels on the chip
```

Comment:
If is this library universal library for MCP3008, the number of channels should be a parameter of the config.

Reply:
Well MCP3008 is always going to have 8 channels. We could move the `channels_mask` param from all the functions below and make it a one-time configuration value though.

Reply:
Moved the mask to the config.